### ISSUE TITLE
add progress bar or similar to `install_cmdstan` and `compile_model` and `sample` functions

### ISSUE THREAD

**Issue by mitzimorris** [stan-dev/cmdstanpy#46]
#### Summary:
Please provide a short couple sentence summary.

#### Description:
Describe the issue as clearly as possible.

#### Additional Information:
Provide any additional information here.

**Comment by maedoc** [comment]
`tqdm` would be the go to package in Python.

I think this would need to run a background thread to periodically parse the csv or stdout for progress.

**Comment by mitzimorris** [comment]
progress bar for sampling added by      https://github.com/stan-dev/cmdstanpy/pull/107

**Comment by mitzimorris** [comment]
for installation and compilation we could display a spinning wheel - cf.
https://pypi.org/project/progress/

but can't really calculate % of task completed as there's no reporting from the compiler.

**Comment by ahartikainen** [comment]
But maybe we could if we parse the output from compiler?

**Comment by mitzimorris** [comment]
how do we know what the compiler plan is?   incremental progress can only be visualized if we know the overall number of steps involved.  the sampler reports % of total because there's a total and it's reporting 100/1000, 200/2000 etc.

**Comment by ahartikainen** [comment]
True, but do we get somehow information what file is currently compiled? Or is that just illusion I usually get reading warnings.

**Comment by mitzimorris** [comment]
make generates a ".d" file which lists the dependencies, but make will only re-compile the dependencies if the source is newer than any existing target.  maybe we could track compiling (which is fast) vs. linking which is not, but it would be a challenge and add a lot of technical debt to the code.

a spinning wheel, however annoying, does indicate that the process is alive and is easy to implement.
